File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Device.ts
import { Annotation } from "./Annotation";
import { CodeableConcept } from "./CodeableConcept";
import { ContactPoint } from "./ContactPoint";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type DeviceDeviceName = BackboneElement & {
    name: string;
    type: string;
};

export type DeviceProperty = BackboneElement & {
    type: CodeableConcept;
    valueCode?: CodeableConcept[];
    valueQuantity?: Quantity[];
};

export type DeviceSpecialization = BackboneElement & {
    version?: string;
    systemType: CodeableConcept;
};

export type DeviceVersion = BackboneElement & {
    type?: CodeableConcept;
    value: string;
    component?: Identifier;
};

export type DeviceUdiCarrier = BackboneElement & {
    issuer?: string;
    entryType?: string;
    carrierHRF?: string;
    carrierAIDC?: string;
    jurisdiction?: string;
    deviceIdentifier?: string;
};

export type Device = DomainResource & {
    patient?: Reference;
    definition?: Reference;
    serialNumber?: string;
    parent?: Reference;
    deviceName?: DeviceDeviceName[];
    property?: DeviceProperty[];
    partNumber?: string;
    modelNumber?: string;
    type?: CodeableConcept;
    statusReason?: CodeableConcept[];
    specialization?: DeviceSpecialization[];
    distinctIdentifier?: string;
    note?: Annotation[];
    status?: string;
    safety?: CodeableConcept[];
    lotNumber?: string;
    url?: string;
    identifier?: Identifier[];
    manufacturer?: string;
    version?: DeviceVersion[];
    location?: Reference;
    contact?: ContactPoint[];
    manufactureDate?: string;
    owner?: Reference;
    expirationDate?: string;
    udiCarrier?: DeviceUdiCarrier[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Device.py
from typing import Optional, List
from base import Annotation
from base import CodeableConcept
from base import ContactPoint
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class Device_DeviceName(BackboneElement):
    name: str
    type: str

class Device_Property(BackboneElement):
    type: CodeableConcept
    value_code: Optional[List[CodeableConcept]] = None
    value_quantity: Optional[List[Quantity]] = None

class Device_Specialization(BackboneElement):
    version: Optional[str] = None
    system_type: CodeableConcept

class Device_Version(BackboneElement):
    type: Optional[CodeableConcept] = None
    value: str
    component: Optional[Identifier] = None

class Device_UdiCarrier(BackboneElement):
    issuer: Optional[str] = None
    entry_type: Optional[str] = None
    carrier_h_r_f: Optional[str] = None
    carrier_a_i_d_c: Optional[str] = None
    jurisdiction: Optional[str] = None
    device_identifier: Optional[str] = None

class Device(DomainResource):
    patient: Optional[Reference] = None
    definition: Optional[Reference] = None
    serial_number: Optional[str] = None
    parent: Optional[Reference] = None
    device_name: Optional[List[Device_DeviceName]] = None
    property: Optional[List[Device_Property]] = None
    part_number: Optional[str] = None
    model_number: Optional[str] = None
    type: Optional[CodeableConcept] = None
    status_reason: Optional[List[CodeableConcept]] = None
    specialization: Optional[List[Device_Specialization]] = None
    distinct_identifier: Optional[str] = None
    note: Optional[List[Annotation]] = None
    status: Optional[str] = None
    safety: Optional[List[CodeableConcept]] = None
    lot_number: Optional[str] = None
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    manufacturer: Optional[str] = None
    version: Optional[List[Device_Version]] = None
    location: Optional[Reference] = None
    contact: Optional[List[ContactPoint]] = None
    manufacture_date: Optional[str] = None
    owner: Optional[Reference] = None
    expiration_date: Optional[str] = None
    udi_carrier: Optional[List[Device_UdiCarrier]] = None